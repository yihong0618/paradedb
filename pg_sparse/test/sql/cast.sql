SELECT ARRAY[1,2,3]::svector;
SELECT ARRAY[1.0,2.0,3.0]::svector;
SELECT ARRAY[1,2,3]::float4[]::svector;
SELECT ARRAY[1,2,3]::float8[]::svector;
SELECT ARRAY[1,2,3]::numeric[]::svector;
SELECT '{NULL}'::real[]::svector;
SELECT '{NaN}'::real[]::svector;
SELECT '{Infinity}'::real[]::svector;
SELECT '{-Infinity}'::real[]::svector;
SELECT '{}'::real[]::svector;
SELECT '{{1}}'::real[]::svector;
SELECT '[1,2,3]'::svector::real[];
SELECT array_agg(n)::svector FROM generate_series(1, 16001) n;
SELECT array_to_svector(array_agg(n), 16001, false) FROM generate_series(1, 16001) n;

-- ensure no error
SELECT ARRAY[1,2,3] = ARRAY[1,2,3];
