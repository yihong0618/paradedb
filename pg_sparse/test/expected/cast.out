SELECT ARRAY[1,2,3]::svector;
  array  
---------
 [1,2,3]
(1 row)

SELECT ARRAY[1.0,2.0,3.0]::svector;
  array  
---------
 [1,2,3]
(1 row)

SELECT ARRAY[1,2,3]::float4[]::svector;
  array  
---------
 [1,2,3]
(1 row)

SELECT ARRAY[1,2,3]::float8[]::svector;
  array  
---------
 [1,2,3]
(1 row)

SELECT ARRAY[1,2,3]::numeric[]::svector;
  array  
---------
 [1,2,3]
(1 row)

SELECT '{NULL}'::real[]::svector;
ERROR:  array must not contain nulls
SELECT '{NaN}'::real[]::svector;
ERROR:  NaN not allowed in svector
SELECT '{Infinity}'::real[]::svector;
ERROR:  infinite value not allowed in svector
SELECT '{-Infinity}'::real[]::svector;
ERROR:  infinite value not allowed in svector
SELECT '{}'::real[]::svector;
ERROR:  svector must have at least 1 dimension
SELECT '{{1}}'::real[]::svector;
ERROR:  array must be 1-D
SELECT '[1,2,3]'::svector::real[];
 float4  
---------
 {1,2,3}
(1 row)

SELECT array_agg(n)::svector FROM generate_series(1, 16001) n;
ERROR:  svector cannot have more than 16000 dimensions
SELECT array_to_svector(array_agg(n), 16001, false) FROM generate_series(1, 16001) n;
ERROR:  svector cannot have more than 16000 dimensions
-- ensure no error
SELECT ARRAY[1,2,3] = ARRAY[1,2,3];
 ?column? 
----------
 t
(1 row)

